crates/task-impls/src/quorum_vote/mod.rs

```diff
@@ -307,6 +361,10 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static, V: Versions> Handl
                         return;
                     }
                     leaf = Some(proposed_leaf);
+                    // Proposed and parent block are the same. We don't care about VID and DAC.
+                    if is_same_block_proposed {
+                        break;
```

Comment:
So here we are in a while loop collecting dependencies, if we break here we possibly won't collect VID or DAC dependencies, then later in line 403 we check we have a vid share, before we vote (same for DAC).  By exiting this loop early we won't get a VID share every time.  

I think the problem is more general though, if at the end of an epoch we don't need a VID share or DAC then this `handle_dep_result` will just never be called.  We will never get the DAC and VID to fullfill the deps and this code will never run.

Are the tests just working because we aren't doing a transition. or because we actually are just resending the DAC or VID?

Reply:
Thanks for the explaining, it makes sense to me know, I didn't realize we still send these out.  I'm now a bit concerned on ordering, It looks like it happens to work because if we get proposal we stop processing, and set everything correct, but process DAC and VID first they have the same "fake" commitment so we don't return early.  Ok for now, but this kind of feels like a house of cards

Reply:
Yeah it works for now, because of that I agree.  I was just saying it feels like this it could easily get messed up if someone who doesn't realize the VID must be overridden for eQC related proposals, deletes that line of code